### ISSUE TITLE
`tap()` Action fails although target item exists

### ISSUE THREAD

**Issue by haitham-reda** [google/EarlGrey#410]
i keep getting `Reason: Action 'Tap' failed` for tapping a `UIButton` subclass although target `UIButton` with the correct `accessibilityId` / `accessibilityLabel` exists and logged in window hierarchy dump after failure.

Runs on iPhone5 simulator, iOS 10.2 , EarlGrey V1.3.0

**Comment by khandpur** [comment]
hi @haitham-reda Can you provide the full error logs (censoring sensitive data) so we can analyze it?

**Comment by tirodkar** [comment]
Also, any reason why you're on EarlGrey 1.3.0?

**Comment by haitham-reda** [comment]
hi @khandpur  , can i forward by Email instead please ?

**Comment by khandpur** [comment]
yes, i've sent you mine @ haitham.reda@gmail.com

**Comment by ArielSD** [comment]
Having the same issue in 1.4.0

**Comment by tirodkar** [comment]
@ArielSD, could you possibly send us some more info such as a log trace or so??

**Comment by ArielSD** [comment]
@tirodkar It doesn't happen every time, next time it does, I'll post!

**Comment by Nebla** [comment]
I'm having a similar issue, that seems to disappear if I run the test with the animations slowed down in the simulator, with regular animations it's failing with this error:

Exception Name: ActionFailedException
Exception Reason: An action failed. Please refer to the error trace below.
Exception with Action: {
  "Action Name" : "Tap",
  "Element Matcher" : "((respondsToSelector(isAccessibilityElement) && isAccessibilityElement) && accessibilityLabel('loginButton'))"
}

Exception Details: Error Trace: [
  {
    "Description" : "Cannot perform action due to a constraint failure.",
    "Domain" : "com.google.earlgrey.ElementInteractionErrorDomain",
    "Code" : "1",
    "File Name" : "GREYBaseAction.m",
    "Function Name" : "-[GREYBaseAction satisfiesConstraintsForElement:error:]",
    "Line" : "69",
    "TestCase Class" : "LoginTests",
    "TestCase Method" : "testLoginWithEmptyFieldsShouldFail"
  }
]

Also, the test appears in a Mac with OS Sierra and Xcode 8.2, with a previous version of Xcode (7.3) seems to work fine.

**Comment by tirodkar** [comment]
@Nebla @haitham-reda @ArielSD this seems extremely interesting, especially since it seems to be an Xcode 8.2 issue. Could you possibly repro this isssue with our FunctionalTests / Contrib / Demo projects please so that we could have a better look?

**Comment by haitham-reda** [comment]
@tirodkar will give it a shot with Xcode 8.3B tomorrow and tell you if any difference.

**Comment by Nebla** [comment]
@tirodkar sorry for the delay, here are the results for functional tests (I renamed from .log to .txt because github does not support .log) using Xcode Version 8.2.1 (8C1002)
[Session-EarlGreyFunctionalTests-2017-02-13_110829-UfTHHM.txt](https://github.com/google/EarlGrey/files/771358/Session-EarlGreyFunctionalTests-2017-02-13_110829-UfTHHM.txt)

I had some problems with the other projects, I'll try to solve them and write the results.

**Comment by tirodkar** [comment]
@Nebla what did you modify the animation speed to? This seems to affect almost every other test in our FunctionalTests which shouldn't happen normally.

**Comment by Nebla** [comment]
Mmm I didn't modify it, I checked out the project and run it as it was on the repo.

**Comment by khandpur** [comment]
@tirodkar From the logs it looks like a keyboard flake. Will https://github.com/google/EarlGrey/pull/327 address it?

**Comment by tirodkar** [comment]
These all do indeed look like flakes that https://github.com/google/EarlGrey/pull/327 should fix. Unfortunately, I've been unable to repro them on local runs. If I could get what animation value was modified, then I'd be able to debug them much better.

**Comment by khandpur** [comment]
I think @Nebla means to say that he didn't modify animation value at all. It's regular speed (1.0), if you down animations on the simulator (using Command + T), the tests start passing. What's concerning is that 48 tests are failing whereas on our travis just a [few 2-3](https://travis-ci.org/google/EarlGrey/jobs/199035583) flake out time-to-time.

**Comment by strongxu** [comment]
I have the same problem. seems If the keyboard is not English keyboard, It will fail. Also if the text is not English, like Chinese, It will also fail.

So can I change keyboard type by EarlGrey?

**Comment by khandpur** [comment]
Can you use a workaround `grey_replaceText:` action until we add support for Chinese keyboard?

Alternatively you can start the app in different language by passing in the scheme the following argument: `-AppleLanguages en`
or programmatically using:
```
[[NSUserDefaults standardUserDefaults] setObject:@"fr" forKey:@"AppleLanguages"];
[[NSUserDefaults standardUserDefaults] synchronize];
```

**Comment by strongxu** [comment]
@khandpur It worked, Thanks a lot!

**Comment by khandpur** [comment]
Closing this issue. Please reopen if you encounter it again.

**Comment by taher-mosbah** [comment]
I have the same issue with an Azerty keyboard.
